tools/istio-iptables/pkg/e2e/e2e_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"testing"
+
+	// Create a new mount namespace.
+	"github.com/howardjohn/unshare-go/mountns"
+	// Create a new network namespace. This will have the 'lo' interface ready but nothing else.
+	_ "github.com/howardjohn/unshare-go/netns"
+	"github.com/howardjohn/unshare-go/userns"
+
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/test/util/assert"
+	cleancmd "istio.io/istio/tools/istio-clean-iptables/pkg/cmd"
+	iptablescmd "istio.io/istio/tools/istio-iptables/pkg/cmd"
+)
+
+func TestIptablesCleanRoundTrip(t *testing.T) {
+	setup(t)
+	// We only have UID 0 in our namespace, so always set it to that
+	const proxyUID = "--proxy-uid=0"
+	t.Run("basic", func(t *testing.T) {
+		assert.NoError(t, runIptables(proxyUID))
+		assert.NoError(t, runIptablesClean(proxyUID))
+		validateIptablesClean(t)
+	})
+	t.Run("dns", func(t *testing.T) {
+		t.Skip("https://github.com/istio/istio/issues/52835")
+		assert.NoError(t, runIptables(proxyUID, "--redirect-dns"))
+		assert.NoError(t, runIptablesClean(proxyUID, "--redirect-dns"))
+		validateIptablesClean(t)
+	})
+}
+
+func validateIptablesClean(t *testing.T) {
+	cur := iptablesSave(t)
+	if strings.Contains(cur, "ISTIO") {
+		t.Fatalf("Istio rules leftover: %v", cur)
```

Comment:
This will likely not be enough to completely validate that iptables are clean.
istio-clean-iptables is also tasked with cleaning non-jump rules in non-istio chains and may leave some leftovers is something is incorrect

e.g:
https://github.com/istio/istio/blob/master/tools/istio-clean-iptables/pkg/cmd/cleanup.go#L76-L80

For the purpose of my e2e cleanup tests I have exposed VerifyIptablesState() to perform this validation
https://github.com/istio/istio/blob/41537135ea7f279cefd0bb646d020721d0432f62/tools/istio-iptables/pkg/capture/run_test.go#L465

Reply:
maybe we can iterate with improved validation

Reply:
These are hard to use directly with this approach since we do not have an `iptConfigurator`.

I have updated it to use the old and new cleanup approach though